src/main/java/htsjdk/samtools/SAMFileHeader.java

```diff
@@ -253,7 +255,12 @@ public SortOrder getSortOrder() {
         if (so == null || so.equals("unknown")) {
             return SortOrder.unsorted;
         }
-        return SortOrder.valueOf((String) so);
+        try {
+            return SortOrder.valueOf(so);
+        } catch (IllegalArgumentException e) {
+            log.warn("Found non conforming header SO tag: "+ so + ". Treating as 'unknown'.");
```

Comment:
The existing implementation here is poor - the fact that it translates from the String on every access.  I hadn't realized it was doing that.  This also means though, that it's going to log your warning on every access.  What do you think about either changing the class to cache the SortOrder enum value _or_ adding `setAttribute("SO", "unknown")` in the catch?